src/include/sof/math/numbers.h

```diff
@@ -12,23 +12,32 @@
 
 #include <stdint.h>
 
+/* These macros evaluate their arguments only once. This comes at a
+ * price:
+ * 1. They require a compiler that supports statement-expressions
+ *    (gcc, clang,...)
+ * 2. They cannot be used at compile time, only at run time.
+ *
+ * Z_MIN and Z_MAX are named like the equivalent Zephyr macros.
+ */
+
 /* Zephyr defines this - remove local copy once Zephyr integration complete */
-#ifdef MIN
-#undef MIN
+#ifdef Z_MIN
```

Comment:
Can we have a sof-rtos.h that defines Z_MIN, Z_MAX etc and also does a
```c
#ifdef ZEPHYR
#error This file should not be in Zephyr build
#endif 
```
This way there is no confusion over which macro we are using.

Reply:
Yes but later please: I want to keep this gigantic commit free of functional changes. At this moment the binary output is the same before/after this commit which makes super quick and convenient to validate; no need to run any test (except for making sure nothing is missing from it, hence the wait for QB).

It also enough to fix https://github.com/zephyrproject-rtos/zephyr/issues/33567 which has been blocked on it for some time (sorry), don't want to delay that more with more complex projects.

Reply:
Now is the best time to make functional changes - as we have just tagged v1.7. We have far bigger problems if we get confused between sof RTS and Zephyr about which macros and APIs we are using.

Reply:
The rest of the PR is good except for this part. 

Today we wrap Zephyr RTOS.
1) Call most SOF RTOS APIs natively.
2) Have a zephyr/wrapper.c that wraps the zephyr calls to look like SOF calls.

The target is to wrap SOF/xtos RTOS
1) Call all RTOS APIs as native zephyr APIs.
2) Have a file and header that wraps the SOF RTOS calls/macros so they look like Zephyr.

This PR needs to create a sof/xtos wrapper.c and header.